Document Agent Telemetry enablement by default (with Agent 7.59)

iglendd opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do? What is the motivation?
<!-- A brief description of the change being made with this pull request. What is your motivation for the PR? -->
Reflect new Agent behavior with 7.59 release

### Merge instructions
It needs to be merged just before Agent 7.59 is released (plan for 10/31/2024 but likely to be released a week or two later)
- [x] Please merge after reviewing

### Additional notes


<!-- Previewing the PR: Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running. -->

olivielpeau commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,74 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+
+{{< site-region region="gov" >}}
+
+Datadog on non-government sites will collect environmental, performance, and feature usage information about the Datadog Agent. When Agent detects a government site or FIPS usage Agent will automatically disable this telemetry collection. When such detection is impossible (for example when proxy is used) Agent telemetry will be emitted but immediately dropped at intake. To avoid this emission in  ...
```
Would be useful to also mention the corresponding env var as an option to disable telemetry (`DD_AGENT_TELEMETRY_ENABLED=false` I guess)

iglendd replied:
Wow, I had no idea that this would automatically work. I test it and it did. Thank you.

iglendd replied:
done

olivielpeau commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,74 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+{{< site-region region="us,us3,us5,eu,ap1" >}}
+Datadog may collect environmental, performance, and feature usage information about the Datadog Agent. This may include diagnostic logs and crash dumps of the Datadog Agent with obfuscated stack traces to support and further improve the Datadog Agent.
+
+You can disable this telemetry collection by updating the `agent_telemetry` setting in the Agent configuration file, as shown in the example below.
```
ditto: I recommend mentioning the env var as well

iglendd replied:
definetly

iglendd replied:
done

olivielpeau commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,74 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+{{% /tab %}}
+{{< /tabs >}}
+
+**Telemetry content:**
```
As discussed live, please add a one-line description about what each metadata and metric represents

iglendd replied:
in process

olivielpeau commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,74 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+- postgres.collect_statement_samples_ms
+- postgres.collect_statement_samples_count
+
+[1]: https://github.com/DataDog/datadog-agent/blob/main/comp/core/agenttelemetry/impl/sender.go#L217-L221
+[2]: https://github.com/DataDog/datadog-agent/blob/main/comp/core/agenttelemetry/impl/config.go#L156
```
Recommendation for these links: link to specific commit hash or tag, so that the link remains valid even after code changes

iglendd replied:
Hmm. did not know about this possibility (but I did feel that line pointing is not good enough). Thank you.

iglendd replied:
done

olivielpeau reviewed (commented):
Left a few comments inline (note: my approval is not needed for this PR to be merged)

iglendd commented:
@hestonhoffman please re-review. Legal and technical content is reviewed and updated. Thank you.

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+
+{{< site-region region="gov" >}}
+
+Agent on non-government sites collects environmental, performance, and feature usage information about the Datadog Agent. When the Agent detects a government site, or [ Datadog Agent FIPS Proxy][1] is used, the Agent automatically disables this telemetry collection. When such detection is impossible (such as, for example, if a proxy is being used) Agent telemetry will be emitted but immediately d ...
```
```suggestion
Agent on non-government sites collects environmental, performance, and feature usage information about the Datadog Agent. When the Agent detects a government site, or the [Datadog Agent FIPS Proxy][1] is used, the Agent automatically disables this telemetry collection. When such detection is impossible (for example, if a proxy is being used), Agent telemetry is emitted, but immediately dropped at Datadog's intake. To avoid this data from being emitted in the first place, Datadog recommends disabling Agent telemetry explicitly by updating the `agent_telemetry` setting in the Agent configuration file, as shown in the example below.
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| Metrics ([source][2])                       | Description                                                                                       |
+| ------------------------------------------- | ------------------------------------------------------------------------------------------------- |
+| checks.execution_time                       | Check's execution time in milliseconds                                                            |
+| pymem.inuse                                 | Number of bytes allocated by the python interpreter                                               |
```
```suggestion
| pymem.inuse                                 | Number of bytes allocated by the Python interpreter                                               |
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| ------------------------------------------- | ------------------------------------------------------------------------------------------------- |
+| checks.execution_time                       | Check's execution time in milliseconds                                                            |
+| pymem.inuse                                 | Number of bytes allocated by the python interpreter                                               |
+| pymem.alloc                                 | Total number of bytes allocated by the python interpreter since the start of the agent            |
```
```suggestion
| pymem.alloc                                 | Total number of bytes allocated by the Python interpreter since the start of the Agent            |
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| checks.execution_time                       | Check's execution time in milliseconds                                                            |
+| pymem.inuse                                 | Number of bytes allocated by the python interpreter                                               |
+| pymem.alloc                                 | Total number of bytes allocated by the python interpreter since the start of the agent            |
+| api_server.request_duration_seconds         | Total number of bytes allocated by the python interpreter since the start of the agent            |
```
```suggestion
| api_server.request_duration_seconds         | Total number of bytes allocated by the Python interpreter since the start of the Agent            |
```
Is this correct? It's the same as the field above

iglendd replied:
@hestonhoffman, missed your question actually ... looking

iglendd replied:
Do not know how I missed that, fixing

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| pymem.alloc                                 | Total number of bytes allocated by the python interpreter since the start of the agent            |
+| api_server.request_duration_seconds         | Total number of bytes allocated by the python interpreter since the start of the agent            |
+| logs.decoded, logs.processed                | CLI commands execution performance (if executed)                                                  |
+| logs.sender_latency                         | Http sender latency in ms                                                                         |
```
```suggestion
| logs.sender_latency                         | HTTP sender latency in milliseconds                                                                         |
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| api_server.request_duration_seconds         | Total number of bytes allocated by the python interpreter since the start of the agent            |
+| logs.decoded, logs.processed                | CLI commands execution performance (if executed)                                                  |
+| logs.sender_latency                         | Http sender latency in ms                                                                         |
+| logs.bytes_missed                           | Total number of bytes lost before they could be consumed by the agent, such as after log rotation |
```
```suggestion
| logs.bytes_missed                           | Total number of bytes lost before they could be consumed by the Agent, such as after log rotation |
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| logs.bytes_missed                           | Total number of bytes lost before they could be consumed by the agent, such as after log rotation |
+| logs.sent                                   | Total number of sent logs                                                                         |
+| logs.dropped                                | Total number of logs dropped                                                                      |
+| logs.bytes_sent                             | Total number of bytes send before encoding if any                                                 |
```
```suggestion
| logs.bytes_sent                             | Total number of bytes send before encoding, if any                                                 |
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| logs.sent                                   | Total number of sent logs                                                                         |
+| logs.dropped                                | Total number of logs dropped                                                                      |
+| logs.bytes_sent                             | Total number of bytes send before encoding if any                                                 |
+| logs.encoded_bytes_sent                     | Total number of sent bytes after encoding if any                                                  |
```
```suggestion
| logs.encoded_bytes_sent                     | Total number of sent bytes after encoding, if any                                                  |
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| logs.dropped                                | Total number of logs dropped                                                                      |
+| logs.bytes_sent                             | Total number of bytes send before encoding if any                                                 |
+| logs.encoded_bytes_sent                     | Total number of sent bytes after encoding if any                                                  |
+| dogstatsd.udp_packets                       | Dogstatsd UDP packets bytes                                                                       |
+| dogstatsd.uds_packets                       | Dogstatsd UDS packets bytes                                                                       |
```
```suggestion
| dogstatsd.udp_packets                       | DogStatsD UDP packets bytes                                                                       |
| dogstatsd.uds_packets                       | DogStatsD UDS packets bytes                                                                       |
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| oracle.statement_plan_errors                | Number of errors in retrieving execution plans                                                    |
+| postgres.collect_relations_autodiscovery_ms | Time to collect autodiscoverty relations in ms                                                    |
+| postgres.collect_stat_autodiscovery_ms      | Time to collect autodiscovery stats in ms                                                         |
+| postgres.get_new_pg_stat_activity_ms        | Time to get pg_stat_activity in ms                                                                |
```
```suggestion
| postgres.get_new_pg_stat_activity_ms        | Time to get `pg_stat_activity` in milliseconds                                                                |
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| oracle.activity_latency                     | Time to retrieve query activity in milliseconds                                                   |
+| oracle.statement_metrics                    | Time to retrieve database metrics in milliseconds                                                 |
+| oracle.statement_plan_errors                | Number of errors in retrieving execution plans                                                    |
+| postgres.collect_relations_autodiscovery_ms | Time to collect autodiscoverty relations in ms                                                    |
+| postgres.collect_stat_autodiscovery_ms      | Time to collect autodiscovery stats in ms                                                         |
```
```suggestion
| postgres.collect_relations_autodiscovery_ms | Time to collect autodiscoverty relations in milliseconds                                                   |
| postgres.collect_stat_autodiscovery_ms      | Time to collect Autodiscovery stats in milliseconds                                                      |
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| postgres.collect_relations_autodiscovery_ms | Time to collect autodiscoverty relations in ms                                                    |
+| postgres.collect_stat_autodiscovery_ms      | Time to collect autodiscovery stats in ms                                                         |
+| postgres.get_new_pg_stat_activity_ms        | Time to get pg_stat_activity in ms                                                                |
+| postgres.get_new_pg_stat_activity_count     | Total rows fetched to collect pg_stat_activity                                                    |
```
```suggestion
| postgres.get_new_pg_stat_activity_count     | Total rows fetched to collect `pg_stat_activity`                                                   |
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| postgres.collect_stat_autodiscovery_ms      | Time to collect autodiscovery stats in ms                                                         |
+| postgres.get_new_pg_stat_activity_ms        | Time to get pg_stat_activity in ms                                                                |
+| postgres.get_new_pg_stat_activity_count     | Total rows fetched to collect pg_stat_activity                                                    |
+| postgres.get_active_connections_ms          | Time to get active connections in ms                                                              |
```
```suggestion
| postgres.get_active_connections_ms          | Time to get active connections in milliseconds                                                              |
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| postgres.get_new_pg_stat_activity_count     | Total rows fetched to collect pg_stat_activity                                                    |
+| postgres.get_active_connections_ms          | Time to get active connections in ms                                                              |
+| postgres.get_active_connections_count       | Total rows fetched to get active connections                                                      |
+| postgres.collect_activity_snapshot_ms       | Time to get activity snapshot in ms                                                               |
+| postgres.collect_statement_samples_ms       | Time to get statement samples in ms                                                               |
```
```suggestion
| postgres.collect_activity_snapshot_ms       | Time to get activity snapshot in milliseconds                                                               |
| postgres.collect_statement_samples_ms       | Time to get statement samples in milliseconds                                                              |
```

hestonhoffman commented on content/en/data_security/agent.md:
```diff
@@ -88,6 +88,100 @@ If you have a requirement to avoid storing secrets in plaintext in the Agent's c
+| postgres.collect_statement_samples_ms       | Time to get statement samples in ms                                                               |
+| postgres.collect_statement_samples_count    | Total rows fetched to collect statement samples                                                   |
+
+Only applicable metrics are emitted. For example, if DBM is not enabled none of the database related metrics are emitted.
```
```suggestion
Only applicable metrics are emitted. For example, if DBM is not enabled, none of the database related metrics are emitted.
```

hestonhoffman reviewed (commented):
Left some style suggestions and a question about one of the table values

iglendd commented:
@hestonhoffman it looks like 7.59 where these feature will be released with will be released this week (Wed or Thu). Can you please merge this PR, thank you

iglendd commented:
@hestonhoffman Ok, I think I understand what happen. I think you have provided two batches of feedback. I have seen and addressed only first. Only now I have realized there was a second batch which I just addressed.

Please double check that your suggestions are applied and merge the PR (I am adding checkbox for the merge too).

Thank you.

hestonhoffman commented:
@iglendd No probs! FYI, we have merge queue in this repo now, so feel free to `/merge` in the future and it should go automatically once it has the required reviews and the checks have passed!

hestonhoffman commented:
/merge

iglendd commented:
Thank you

hestonhoffman commented:
@olivielpeau, looks like this needs an approval from you as well
